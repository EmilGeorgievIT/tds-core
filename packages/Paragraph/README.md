# TDS Core: Paragraph

Block text as an HTML `<p>` element.
