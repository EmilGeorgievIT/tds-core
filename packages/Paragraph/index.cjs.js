require('./dist/index.css')
const Paragraph = require('./dist/index.cjs')

module.exports = Paragraph
