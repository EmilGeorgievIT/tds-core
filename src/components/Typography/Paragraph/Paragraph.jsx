import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import safeRest from '../../../safeRest'

import styles from './Paragraph.modules.scss'


const Paragraph = ({ bold, size, children, ...rest }, context) => {
  const classes = classnames(
    context.inheritColor ? styles.inheritColor : styles.color,
    styles[size],
    {
      [styles.bold]: bold
    }
  )

  return (
    <p {...safeRest(rest)} className={classes}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  bold: PropTypes.bool,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large'
  ]),
  children: PropTypes.node.isRequired
}

Paragraph.defaultProps = {
  bold: false,
  size: 'medium'
}

Paragraph.contextTypes = {
  inheritColor: PropTypes.bool
}

export default Paragraph
