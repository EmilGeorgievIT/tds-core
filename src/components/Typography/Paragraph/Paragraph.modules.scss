@import '../../../scss/settings/colours';

.base {
  padding: 0;
  margin: 0;

  text-align: left;

  &.bold {
    font-weight: 700; // Neue Medium 65
  }
}

.small {
  composes: base;

  font-weight: 500; // Neue Roman 55
  font-size: .875rem;
  letter-spacing: -0.6px;
  line-height: 1.5;
}

.medium {
  composes: base;

  font-weight: 400; // neue 45 light
  font-size: 1rem;
  letter-spacing: -0.8px;
  line-height: 1.5;
}

.large {
  composes: base;

  font-weight: 400; // neue 45 light
  font-size: 1.25rem;
  letter-spacing: -1px;
  line-height: 2;
}

.color {
  color: $color-text;
}

.inheritColor {
  color: inherit;
}
