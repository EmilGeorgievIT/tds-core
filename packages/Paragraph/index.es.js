import './dist/index.css'
import Paragraph from './dist/index.es'

export default Paragraph
