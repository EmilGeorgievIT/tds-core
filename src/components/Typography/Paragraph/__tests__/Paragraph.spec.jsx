import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Paragraph from '../Paragraph'

describe('Paragraph', () => {
  const doShallow = props => (
    shallow(<Paragraph {...props}>Some content</Paragraph>)
  )

  it('renders', () => {
    const paragraph = doShallow()

    expect(toJson(paragraph)).toMatchSnapshot()
  })

  it('renders an HTML p tag', () => {
    const paragraph = doShallow()

    expect(paragraph).toHaveTagName('p')
  })

  it('can be bold', () => {
    let paragraph = doShallow()
    expect(paragraph).not.toHaveClassName('bold')

    paragraph = doShallow({ bold: true })
    expect(paragraph).toHaveClassName('bold')
  })

  it('can be sized', () => {
    let paragraph = doShallow()
    expect(paragraph).toHaveClassName('medium')

    paragraph = doShallow({ size: 'small' })
    expect(paragraph).toHaveClassName('small')
  })

  it('passes additional attributes to the p element', () => {
    const paragraph = doShallow({ id: 'the-paragraph' })

    expect(paragraph).toHaveProp('id', 'the-paragraph')
  })

  it('does not allow custom CSS', () => {
    const paragraph = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(paragraph).not.toHaveProp('className', 'my-custom-class')
    expect(paragraph).not.toHaveProp('style')
  })
})
