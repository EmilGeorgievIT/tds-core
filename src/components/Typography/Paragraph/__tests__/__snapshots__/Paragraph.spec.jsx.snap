// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders 1`] = `
<p
  className="color medium"
>
  Some content
</p>
`;
